sql/core/src/main/scala/org/apache/spark/sql/api/r/SQLUtils.scala

```diff
@@ -225,4 +225,10 @@ private[sql] object SQLUtils extends Logging {
     }
     sparkSession.sessionState.catalog.listTables(db).map(_.table).toArray
   }
+
+  def createArrayType(elementType: DataType): ArrayType = DataTypes.createArrayType(elementType)
+
+  def createArrayType(elementType: Column): ArrayType = {
+    new ArrayType(ExprUtils.evalTypeExpr(elementType.expr), true)
```

Comment:
Instead of `true`, can we use `elementType.expr.nullable`?

Reply:
Yup, that sounds more correct.

Reply:
Oops, it looks not actually. `elementType.expr.nullable` will return nullability from the expression (like `schema_of_json` or string literal), not for something related with its input schema.